app/cards/[id]/page.tsx

```diff
@@ -66,10 +66,15 @@ export default function CardItem() {
         Вкажіть вагу та зріст Вашої дитини по місяцям для побудови графіку
       </h2>
       <Form id={cardId} records={records} />
-      <IndicatorsTable id={cardId} records={records} />
-      <Button type="submit" className="mb-6 mt-5" onClick={goToChart}>
-        Побудувати графік
-      </Button>
+
+      {records.length > 0 && (
+        <div>
+          <IndicatorsTable id={cardId} records={records} />
+          <Button type="submit" className="mb-6 mt-5" onClick={goToChart}>
+            Побудувати графік
+          </Button>
```

Comment:
There's no need to use `type=submit` here. Also, I suggest you to use `NextLink` instead of `router.push`. See `line 49` as an example. 
```suggestion
          <Link href={`/cards/${cardId}/charts`} className="mb-6 mt-5">
            <Button>
              Побудувати графік
            </Button>
          </Link>            
```